reps/2023-03-20-air-storage-path.md

```diff
@@ -0,0 +1,151 @@
+in the configuration.
+
+Risk: Since we will continue to use locally cached data for the implementation,
+hiding the location in e.g. an environment variable can make failures more opaque
+to users.
+
+
+### Should this change be within `ray` or outside?
+
+This is an API change to Ray AIR (specifically, `air.RunConfig`, `tune.SyncConfig`, and `tune.run()`)
+and therefore has to be within Ray.
+
+
+## Stewardship
+
+### Required Reviewers
+
+@richardliaw, @ericl
+
+### Shepherd of the Proposal (should be a senior committer)
+
+@richardliaw
+
+## Design and Architecture
+
+- We introduce a new `storage_path` argument to `air.RunConfig` and `tune.run` as the main configuration entrypoints
+- If the `storage_path` is set to a remote URI, the `local_path` is read from an environment variable `RAY_AIR_CACHE_DIR`
+- Backwards compatibility: If a `local_dir` is passed, we set this environment variable
+- In downstream components (`TrialRunner`, `Experiment`, `Trial`), we introduce respective arguments: `storage_path` and `experiment_path`
+
+We retain full backwards compatibility with the current API.
+
+## Usage Example
+
+### Basic usage
+
+```python
+import ray
+from ray import air, tune
+
+results = tune.Tuner(train_fn, run_config=air.RunConfig(storage_path="s3://foo/bar")).fit()
+assert results.best_checkpoint().path.startswith("s3://foo/bar")
+```
+
+### Usage when Ray storage is configured
+
+```python
+import ray
+from ray import air, tune
+
+# STORAGE_PATH is configured to s3://foo/bar
+
+results = tune.Tuner(train_fn).fit()
+assert results.best_checkpoint().path.startswith("s3://foo/bar")
+```
+
+### Accessing storage paths
+
+```python
+import ray
+from ray import tune
+
+results = tune.Tuner(train_fn).fit()
+
+assert results.path.startswith("s3://bucket/dir/experiment_dir")
+
+result = results.get_best_result()
+assert result.path.startswith("s3://bucket/dir/experiment_dir/trial0")
+
+checkpoint = result.checkpoint
+assert checkpoint.path.startswith("s3://bucket/dir/experiment_dir/trial0/checkpoint_0010")
+```
+
+## Compatibility, Deprecation, and Migration Plan
+
+### Compatibility
+
+We retain full backwards compatibility with the current API.
+
+The initial PR merges will keep the old API in the unit tests.
+
+### Deprecation plan
+
+We are changing stable public APIs. Thus, we will follow a deprecation plan over three releases:
+
+For public APIs (`tune.Tuner()`, `tune.run()`):
+
+- In Ray 2.4, we soft deprecate `RunConfig.local_dir` and `SyncConfig.upload_dir` and warn if they are used
+- We keep this behavior in Ray 2.5
+- In Ray 2.6 we will raise an error if those arguments are still used
+- In Ray 2.7 we remove the old code paths
+
+For developer APIs (`Trial`, `TrialRunner`, `Experiment`) we can speed up hard deprecation
+by one release.
+
+#### Syncer class
+
+Currently, when a `local_dir` but no `upload_dir` is specified, we synchronize trial data (e.g. checkpoints)
```

Comment:
I think the end behavior here needs to be part of the REP.

Reply:
removed this section with respect to the discussion in the REP